controllers/snapshot/snapshot_adapter.go

```diff
@@ -154,6 +154,7 @@ func (a *Adapter) EnsureGlobalCandidateImageUpdated() (reconciler.OperationResul
 		for _, component := range a.snapshot.Spec.Components {
 			if component.Name == a.component.Name {
 				a.component.Spec.ContainerImage = component.ContainerImage
+				a.component.Status.LastBuiltCommit = component.Source.GitSource.Revision
```

Comment:
In order to update the component's status, we will need to also call `a.client.Status.Patch` separately, like e.g. [here](https://github.com/redhat-appstudio/integration-service/blob/main/gitops/snapshot.go#L118).

Reply:
Thank you, that makes sense.
Just out of curiosity, wouldn't this already be taken care of by `a.client.Patch()`? It would update both the `spec` and `status` sections of the Component together?

Reply:
I thought so too initially but it turned out It wouldn't unfortunately

Reply:
We'll also need to check if the `component.Source.GitSource` is defined before using it here. This should resolve the [unit test failure](https://github.com/redhat-appstudio/integration-service/actions/runs/4600217515/jobs/8126517849?pr=141).

Reply:
Thanks for your help!